docs/installing-solidity.rst

```diff
@@ -174,11 +174,12 @@ Install it using ``brew``:
     # eg. Install 0.4.8
     brew install https://raw.githubusercontent.com/ethereum/homebrew-ethereum/77cce03da9f289e5a3ffe579840d3c5dc0a62717/solidity.rb
 
-Gentoo Linux also provides a solidity package that can be installed using ``emerge``:
+Gentoo Linux has an overlay at https://overlays.gentoo.org/#ethereum that contains a solidity package.
+After the overlay is setup, ``solc`` can be installed in x86_64 architectures by:
 
 .. code-block:: bash
 
-    emerge dev-lang/solidity
+    ACCEPT_KEYWORDS='~amd64' emerge dev-lang/solidity
```

Comment:
Is the manual selection of amd64 necessary?

Reply:
`dev-lang/solidity` was marked unstable in the ethereum overlay as it depends on other packages that are currently unstable (particularly, `sci-mathematics/cvc4`).

In Gentoo, the quickest way to unmask unstable packages for installation in the x86_64 architecture is by adding `~amd64` (with the tilde) to the `ACCEPT_KEYWORDS` environment variable.

Let me know if I should elaborate a bit about the unstable-ness state of this package.

Reply:
I don't know much about gentoo, but cvc4 should just be an optional dependency.

Reply:
@chriseth: cvc4 is an optional dependency, as well as z3. For example, the following command compiles `solc`  without cvc4 or z3 support in a Gentoo box with the ethereum overlay:
```
ACCEPT_KEYWORDS='~amd64' USE='-cvc4 -z3' emerge dev-lang/solidity
```
This does not change the fact that `dev-lang/solidity` should be considered an unstable package according to Gentoo packaging good practices, as it **may** depend on the unstable-only cvc4 according to which USE flags are exercised at the time of installation.

If you guys believe it pays, I'm ok with describing all this in some wiki at the [gentoo ethereum overlay](https://github.com/coolparadox/ethereum-gentoo-overlay) and update my PR in order the installation doc links to it.

Reply:
I don't understand very much about gentoo packaging, but using such an environment variable to force installation of unstable packages seems rather arcane to me :)

How "bad" is it for a package to be unstable? Should we rather remove CVC4 such that the package can be stable?

Reply:
> I don't understand very much about gentoo packaging, but using such an environment variable to force installation of unstable packages seems rather arcane to me :)

Yeah that's Gentoo Linux... Optionally we can edit configuration files under /etc and set per-package flags permanently. It's a bit hardcore, but not surprising in a distro that instructs the user to compile his/her own kernel as part of the installation process ;)

> How "bad" is it for a package to be unstable?

From my experience, the Gentoo maintainers like to mark the most recent major versions of anything as 'unstable'. It's no big deal actually; in my system I have tenths of 'unstable' packages installed and everything works despite -- in some cases, because -- of it.

> Should we rather remove CVC4 such that the package can be stable?

Done :)

Reply:
Thanks!